langroid/embedding_models/models.py

```diff
@@ -139,6 +171,72 @@ def embedding_dims(self) -> int:
         return self.config.dims
 
 
+class AzureOpenAIEmbeddings(EmbeddingModel):
+    """
+    Azure OpenAI embeddings model implementation.
+    """
+
+    def __init__(
+        self, config: AzureOpenAIEmbeddingsConfig = AzureOpenAIEmbeddingsConfig()
+    ):
+        """
+        Initializes Azure OpenAI embeddings model.
+
+        Args:
+            config: Configuration for Azure OpenAI embeddings model.
+        Raises:
+            ValueError: If required Azure config values are not set.
+        """
+        super().__init__()
+        self.config = config
+        load_dotenv()
+
+        self.config.api_key = os.getenv("AZURE_OPENAI_API_KEY", "")
+        if self.config.api_key == "":
+            raise ValueError(
+                """AZURE_OPENAI_API_KEY env variable must be set to use 
+            AzureOpenAIEmbeddings. Please set the AZURE_OPENAI_API_KEY value 
+            in your .env file."""
+            )
+        self.config.azure_endpoint = os.getenv("AZURE_OPENAI_ENDPOINT", "")
```

Comment:
can you please change it to:
````
        if not self.config.azure_endpoint:
            self.config.azure_endpoint = os.getenv("AZURE_OPENAI_ENDPOINT", "")
````
so that if user passes azure_endpoint in Config object it's not overwritten?